libs/scalalib/src/mill/scalalib/CoursierConfigModule.scala

```diff
@@ -0,0 +1,180 @@
+    (env, props)
+  }
+
+  /** Default repositories for dependency resolution */
+  def defaultRepositories: Task[Seq[Repository]] = Task.Anon {
+    val (env, props) = coursierEnv()
+    CoursierEnv.defaultRepositories(
+      CoursierEnv.repositories.readFrom(env, props),
+      CoursierEnv.scalaCliConfig.readFrom(env, props)
+    )
+  }
+
+  /** Default JSON configuration files to be used by coursier */
+  def defaultConfFiles: Task[Seq[PathRef]] = Task.Anon {
+    val (env, props) = coursierEnv()
+    CacheEnv.defaultConfFiles(CacheEnv.scalaCliConfig.readFrom(env, props))
+      .map(os.Path(_, BuildCtx.workspaceRoot))
+      .map(PathRef(_))
+  }
+
+  /** Default mirrors to be used by coursier */
+  def defaultMirrors: Task[Seq[Mirror]] = Task.Anon {
+    val (env, props) = coursierEnv()
+    CoursierEnv.defaultMirrors(
+      CoursierEnv.mirrors.readFrom(env, props),
+      CoursierEnv.mirrorsExtra.readFrom(env, props),
+      CoursierEnv.scalaCliConfig.readFrom(env, props),
+      CoursierEnv.configDir.readFrom(env, props)
+    )
+  }
+
+  /** Default cache location to be used by coursier */
+  def defaultCacheLocation: Task[String] = Task.Anon {
+    val (env, props) = coursierEnv()
+    val path: java.nio.file.Path = CacheEnv.defaultCacheLocation(
+      CacheEnv.cache.readFrom(env, props)
+    )
+    path.toString
+  }
+
+  /**
+   * Default archive cache location to be used by coursier
+   *
+   * This is the directory under which archives are unpacked
+   * by `coursier.cache.ArchiveCache`. This includes JVMs
+   * managed by coursier, which Mill relies on by default.
+   */
+  def defaultArchiveCacheLocation: Task[String] = Task.Anon {
+    val (env, props) = coursierEnv()
+    val path: java.nio.file.Path = CacheEnv.defaultArchiveCacheLocation(
+      CacheEnv.archiveCache.readFrom(env, props)
+    )
+    path.toString
+  }
+
+  /** Default repository credentials to be used by coursier */
+  def defaultCredentials: Task[Seq[Credentials]] = Task.Anon {
+    val (env, props) = coursierEnv()
+    CacheEnv.defaultCredentials(
+      CacheEnv.credentials.readFrom(env, props),
+      CacheEnv.scalaCliConfig.readFrom(env, props),
+      CacheEnv.configDir.readFrom(env, props)
+    )
+  }
+
+  /** Default TTL used by coursier when downloading changing artifacts such as snapshots */
+  def defaultTtl: Task[Option[Duration]] = Task.Anon {
+    val (env, props) = coursierEnv()
+    CacheEnv.defaultTtl(
+      CacheEnv.ttl.readFrom(env, props)
+    )
+  }
+
+  /** Default cache policies used by coursier when downloading artifacts */
+  def defaultCachePolicies: Task[Seq[CachePolicy]] = Task.Anon {
+    val (env, props) = coursierEnv()
+    CacheEnv.defaultCachePolicies(
+      CacheEnv.cachePolicy.readFrom(env, props)
+    )
+  }
+
+  /**
+   * Aggregates coursier parameters read from the environment
+   */
+  def coursierConfig: Task[mill.util.CoursierConfig] = Task.Anon {
```

Comment:
Looks like a candidate for caching, to me.
```suggestion
  def coursierConfig: Task[mill.util.CoursierConfig] = Task {
```

Reply:
Let's leave it as-is then. Would be nice to have some support for secret caching (e.g. via encryption, or just in memory), but we currently don't.